Probe/src/probe/rpc.go

```diff
@@ -0,0 +1,142 @@
+/*
+ *  Copyright 2020 Google LLC
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package probe
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/FirebaseExtended/fcm-external-prober/Controller/src/controller"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+var client controller.ProbeCommunicatorClient
+var pingConfig *controller.PingConfig
+var hostname string
+var retries int32
+
+func initClient() error {
+	tls, err := credentials.NewClientTLSFromFile("cert.pem", "")
+	if err != nil {
+		return err
+	}
+	//TODO: determine through which mode the internal DNS address of the controller will be available
+	//Might be best to just throw it in metadata with the cert
+	conn, err := grpc.Dial("crdhost.c.gifted-cooler-279818.internal:50001", grpc.WithTransportCredentials(tls), grpc.WithBlock(), grpc.WithTimeout(15 * time.Second))
+	if err != nil {
+		return err
+	}
+	client = controller.NewProbeCommunicatorClient(conn)
+
+	hostname, err = getHostname()
+	if err != nil {
+		return err
+	}
+
+	c := make(chan *controller.RegisterResponse)
+	fmt.Println("Register")
+	go register(c)
+
+	cfg := <- c
+	fmt.Println("Done Register")
```

Comment:
Make this into a log message if it's still needed

Reply:
Addressed in 282c3d0